lib/fizzy/execute.cpp

```diff
@@ -139,15 +139,20 @@ inline uint32_t grow_memory(
     bytes& memory, uint32_t delta_pages, uint32_t memory_pages_limit) noexcept
 {
     const auto cur_pages = memory.size() / PageSize;
-    assert(cur_pages <= size_t(std::numeric_limits<int32_t>::max()));
-    const auto new_pages = cur_pages + delta_pages;
-    assert(new_pages >= cur_pages);
+    // Guaranteed by allocation in instantiate and this function for subsequent increases.
+    assert(memory.size() % PageSize == 0);
+    assert(memory_pages_limit <= MaxMemoryPagesLimit);
+    assert(cur_pages <= MaxMemoryPagesLimit);
+
+    const auto new_pages = uint64_t{cur_pages} + delta_pages;
     if (new_pages > memory_pages_limit)
         return static_cast<uint32_t>(-1);
 
     try
     {
-        memory.resize(new_pages * PageSize);
+        // new_pages <= memory_pages_limit <= MaxMemoryPagesLimit guarantees multiplication
```

Comment:
Not sure we need this entire comment now with the asserts above?

Reply:
To explain why we are sure it's safe not to check for overflow here?

Reply:
I mean now the `uint64_t{}` conversion and the assertions explain it well enough?

Reply:
I would leave it, it's like an inferential step between assertions and conlusion that overlow is not possible. And it shows that we didn't forget to think about it :)

Reply:
Actually would we need a `static_assert(sizeof(size_t) >= sizeof(uint32_t));` here for total clarity? 🙊